demo/with-cra/src/components/NavItem.tsx

```diff
@@ -0,0 +1,51 @@
+import styled from '@emotion/styled';
+import React from 'react';
+import {NavLink} from 'react-router-dom';
+import {Li} from './ProductNav';
+
+export interface NavItemProps {
+  href: string;
+  children: React.ReactNode;
+  onClick?: (e: React.MouseEvent) => void;
+  isActive: boolean;
+}
+export const NavItem = ({href, children, onClick, isActive}: NavItemProps) => {
+  const handleClick = (e: React.MouseEvent) => {
+    onClick?.(e);
+  };
+
+  return (
+    <Li>
+      <NavItemButton
```

Comment:
react-router-dom 라이브러리에서 제공해주는 NavLink style 의 props 인수로 acitve 상태를 확인하고 있는게 아니라
isActive 라는 **일반 변수**로 acitve 상태를 판별하는거라면 style 코드를 jsx에 안 적어도 좋을 것 같아요
일반적인 style 관련된 코드는 emotion 라이브러리를 통해 작성하는 것이 좋을 것 같습니다. html css 코드가 분리되어서 더 가독성에도 좋구요 ㅎㅎ

```suggestion
      <NavItemButton
        to={href}
        onClick={handleClick}
        isActive={isActive}
      >
        {children}
      </NavItemButton>
      
      
const NavItemButton = styled(NavLink)<{isActive: boolean}>`
  ...
  
  ${props =>
    props.isActive &&
    css`
      color: #0095f6;
      background-color: #f7f7f7 !important;
    `}
`;
```